### PR TITLE
Add new case iozone_windows.with_gpt_disk

### PR DESCRIPTION
**Pull request by peixiu**
Add a new case to do iozone test for gpt format disk.

id: 1566836
Signed-off-by: Peixiu Hou <phou@redhat.com>

### CODE REVIEW

**Comment by peixiu** [comment]
@kimi1978wy could you help to review this patch?
thanks~

**Comment by peixiu** [comment]
@YongxueHong could you help to review this patch? thanks a lot~

File: generic/tests/cfg/iozone_windows.cfg [hunk 1]
```diff
@@ -29,3 +29,13 @@
             force_create_image_stg1 = yes
             remove_image_stg0 = yes
             remove_image_stg1 = yes
+        - gpt_disk_test:
```
Comment (YongxueHong) [inline generic/tests/cfg/iozone_windows.cfg]: How about the case name `with_gpt_disk`? the word "test" covers the case `iozone_windows`.

Comment (peixiu) [reply generic/tests/cfg/iozone_windows.cfg]: that's ok, will update, thanks~

File: generic/tests/iozone_windows.py [hunk 2]
```diff
@@ -86,6 +87,15 @@ def run_iozone_parallel(timeout):
 
         logging.info('All the iozone threads are done.')
 
+    def check_gpt_labletype(disk_index):
+        """
+        """
```
[the remark this answers is not in the document]
Comment (peixiu) [reply generic/tests/iozone_windows.py]: ok, will update~

File: generic/tests/iozone_windows.py [hunk 3]
```diff
@@ -86,6 +87,15 @@ def run_iozone_parallel(timeout):
...
+        cmd = "echo list disk > test.dp & diskpart /s test.dp && del test.dp"
+        pattern = r'Disk %s.+?B\s{8}\*' % disk_index
+        if re.search(pattern, session.cmd_output(cmd)):
+            return True
+        return False
```
[the remark this answers is not in the document]
Comment (peixiu) [reply generic/tests/iozone_windows.py]: thanks, will updated it

**Comment by peixiu** [comment]
(1/1) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.iozone_windows.with_gpt_disk.q35: PASS (387.16 s)

**Comment by peixiu** [comment]
(1/1) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.i386.sp1.io-github-autotest-qemu.iozone_windows.with_gpt_disk.q35: PASS (1690.82 s)

**Comment by kimi1978wy** [comment]
The new patch solved the error "FAIL: Disk partition type is not gpt" problem for win7 guests.
LGTM now
(1/1) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.iozone_windows.with_gpt_disk.q35: PASS (384.53 s)
 (1/1) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.iozone_windows.with_gpt_disk.q35: PASS (316.82 s)

**Comment by peixiu** [comment]
@YongxueHong could you help to review it again?
Thanks a lot~

**Comment by peixiu** [comment]
(1/1) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.iozone_windows.with_gpt_disk.q35: PASS (385.14 s)
 (1/1) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.i386.sp1.io-github-autotest-qemu.iozone_windows.with_gpt_disk.q35: PASS (1887.24 s)

**Comment by peixiu** [comment]
@YongxueHong @kimi1978wy I update the code, could you help to review it again? 
thanks a lot~

**Comment by peixiu** [comment]
based on  https://github.com/avocado-framework/avocado-vt/pull/2191

File: generic/tests/iozone_windows.py [hunk 4]
```diff
@@ -103,18 +113,27 @@ def run_iozone_parallel(timeout):
...
                                                                 test, driver_name,
                                                                 timeout)
+
     if params.get("format_disk", "no") == "yes":
         error_context.context("Format disk", logging.info)
         for index, letter, fstype in zip(disk_indexes, disk_letters, disk_fstypes):
-            utils_misc.format_windows_disk(session, index, letter, fstype=fstype)
+            utils_misc.format_windows_disk(session, index, letter, fstype=fstype,
+                                           labletype=labletype)
+
```
[the remark this answers is not in the document]
Comment (peixiu) [reply generic/tests/iozone_windows.py]: for this case, all guests will create to GPT partition table firstly~

**Comment by XueqiangWei** [comment]
(1/1) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.iozone_windows.with_gpt_disk.q35: PASS (536.86 s)

LGTM,
Acked-by xuwei@redhat.com

**Comment by YongxueHong** [comment]
LGTM

**Pull request merged**